### PR TITLE
redesigned the antiflood method for guaranteed message delivery

### PR DESCRIPTION
**Pull request by alex75311**
## Description
redesigned the antiflood method for guaranteed message delivery

## Checklist:
- [X] I added/edited example on new feature/change (if exists)
- [X] My changes won't break backward compatibility
- [X] I made changes both for sync and async

### CODE REVIEW

**Comment by coder2020official** [comment]
Hi!
Thank you for PR. We'll review it after bot api 6.6 update

**Comment by alex75311** [comment]
the library uses multithreading. another thread manages to squeeze in and send, and part of the threads returns error 429 with a new timeout. my solution guarantees that the message will be sent\modified.

**Review by Badiboy — APPROVED** [review]

**Comment by coder2020official** [comment]
Should be tested first

**Comment by coder2020official** [comment]
the original function does not raise an error when the message is not delivered, i don't see any reason for redesigning cuz the previous version already handles 429 error which is the only error to be handled

**Comment by alex75311** [comment]
above I described a case in which the current solution did not work for me, so I changed it to a solution that delivers 100% messages.

File: telebot/util.py [hunk 1]
```diff
@@ -613,14 +613,14 @@ def antiflood(function: Callable, *args, **kwargs):
...
     from time import sleep
 
-    try:
-        return function(*args, **kwargs)
-    except ApiTelegramException as ex:
-        if ex.error_code == 429:
-            sleep(ex.result_json['parameters']['retry_after'])
+    while True:
+        try:
             return function(*args, **kwargs)
```
Comment (alex75311) [inline telebot/util.py:618]: if error 429 is returned here again, then it is not processed in any way

**Comment by coder2020official** [comment]
But raising the error is a breaking change

**Comment by Badiboy** [comment]
> But raising the error is a breaking change

It raised error before. With this update it will raise it more rare.

From my side I cannot accept infinite loops in processing. Various things may happen, thus infinite loop will bring to stall. Loop may exist but it should be finite. May be with configurable (by parameter) number of retries.
@alex75311 👆

**Comment by alex75311** [comment]
I added 5 attempts, that should be enough

**Comment by Badiboy** [comment]
@coder2020official Looks fine for me now.

**Comment by Badiboy** [comment]
@coder2020official Reasonable.

@alex75311 Please, set in line 628 "raise" instead of "return ...".

**Comment by coder2020official** [comment]
i am talking about last line:
image
why is an error being raised? is it intended?

**Comment by alex75311** [comment]
if we are talking about line 6 of your screenshot, then this exception is called if the function returned an error other than 429

**Comment by coder2020official** [comment]
Yes, is it intended? Before it never raised an error

**Comment by alex75311** [comment]
look carefully at how it was before

**Comment by Badiboy** [comment]
> Yes, is it intended? Before it never raised an error

image

Absolutely the same.

**Comment by coder2020official** [comment]
Fine then, I probably skipped the last part

**Review by coder2020official — APPROVED** [review]

**Pull request merged**